src/waveform/waveformwidgetfactory.cpp

```diff
@@ -579,7 +583,17 @@ void WaveformWidgetFactory::setDisplayBeatGridAlpha(int alpha) {
     for (std::size_t i = 0; i < m_waveformWidgetHolders.size(); i++) {
         m_waveformWidgetHolders[i].m_waveformWidget->setDisplayBeatGridAlpha(m_beatGridAlpha);
     }
+}
+
+void WaveformWidgetFactory::setBeatGridMode(int mode) {
+    m_beatGridMode = mode;
+    if (m_waveformWidgetHolders.size() == 0) {
+        return;
+    }
```

Comment:
Is this check necessary? Shouldn't the loop below go through 0 iterations if m_waveformWidgetHolders is empty?